Release Notes/Release Notes v2.11.0-beta1.md

```diff
@@ -0,0 +1,42 @@
+# .NET Driver Version 2.11.0-beta1 Release Notes
+
+This is a beta release for the 2.11.0 version of the driver.
+
+The main new features in 2.11.0-beta1 are:
```

Comment:
Let's mention that the 2.11 release is intended to support MongoDB 4.4, and touch on the small API changes that are part of that.

Reply:
Gotcha, I'll update this next!

Reply:
Updated!